app/pages/nonsipp/employercontributions/EmployerTypeOfBusinessPage.scala

```diff
@@ -17,16 +17,31 @@
 package pages.nonsipp.employercontributions
 
 import config.Refined.{Max300, Max50}
-import models.IdentityType
+import models.{IdentityType, UserAnswers}
 import models.SchemeId.Srn
 import pages.QuestionPage
 import play.api.libs.json.JsPath
 import utils.RefinedUtils.RefinedIntOps
 
+import scala.util.Try
+
 case class EmployerTypeOfBusinessPage(srn: Srn, memberIndex: Max300, index: Max50) extends QuestionPage[IdentityType] {
 
   override def path: JsPath =
     Paths.memberEmpContribution \ toString \ memberIndex.arrayIndex.toString \ index.arrayIndex.toString
 
   override def toString: String = "orgType"
+
+  override def cleanup(value: Option[IdentityType], userAnswers: UserAnswers): Try[UserAnswers] =
+    (value, userAnswers.get(this)) match {
+      case (None, _) => Try(userAnswers)
```

Comment:
Sorry for asking this before it's ready for review, but should the cleanup happen in this case, i.e. `case (None, _)`? If kept on default, it will be called on a change too, right?

Reply:
no worries, what do you mean by `If kept on default, it will be called on a change too, right?`

Reply:
Sorry, I'll explain what I meant.
Currently  default case `case _` is executed in case of new records, removal and change to a different values.
I was thinking about something like this:
```
    (value, userAnswers.get(this)) match {
      case (None, _) => Try(userAnswers) // delete handled by cleanup in EmployerNamePage
      case (Some(_), None) => Try(userAnswers) // new value no need to delete anything
      case (Some(IdentityType.UKCompany), Some(IdentityType.UKCompany)) => Try(userAnswers)
      case (Some(IdentityType.UKPartnership), Some(IdentityType.UKPartnership)) => Try(userAnswers)
      case (Some(IdentityType.Other), Some(IdentityType.Other)) => Try(userAnswers)
      case _ =>
        userAnswers
          .remove(EmployerCompanyCrnPage(srn, memberIndex, index))
          .flatMap(_.remove(PartnershipEmployerUtrPage(srn, memberIndex, index)))
          .flatMap(_.remove(OtherEmployeeDescriptionPage(srn, memberIndex, index)))
    }
```